molter/help.py

```diff
@@ -0,0 +1,197 @@
+        run_checks: bool = False,
+        show_self: bool = False,
+        show_params: bool = False,
+        show_aliases: bool = False,
+        show_prefix: bool = False,
+        embed_title: str | None = None,
+        not_found_message: str | None = None,
+    ) -> None:
+        self._client = client
+        self.show_hidden = show_hidden
+        self.run_checks = run_checks
+        self.show_self = show_self
+        self.show_params = show_params
+        self.show_aliases = show_aliases
+        self.show_prefix = show_prefix
+        self.embed_title = embed_title or "{username} Help Command"
+        self.not_found_message = not_found_message or "Sorry! No command called `{cmd_name}` was found."
+        self.cmd = self._callback
+
+    def register(self) -> None:
+        """Register the help command in dis-snek."""
+        if not isinstance(self.cmd.callback, functools.partial):
+            # prevent wrap-nesting
+            self.cmd.callback = functools.partial(self.cmd.callback, self)
+
+        # replace existing help command if found
+        if "help" in self._client.commands:
+            log.warning("Replacing existing help command.")
+            del self._client.commands["help"]
+
+        self._client.add_message_command(self.cmd)  # type: ignore
+
+    async def send_help(self, ctx: dis_snek.MessageContext, cmd_name: str | None) -> None:
+        """
+        Send a help message to the given context.
+
+        args:
+            ctx: The context to use
+            cmd_name: An optional command name to send help for.
+        """
+        await self._callback.callback(ctx, cmd_name)  # type: ignore
+
+    @molter.msg_command(name="help")
+    async def _callback(self, ctx: dis_snek.MessageContext, cmd_name: str = None) -> None:
+        if cmd_name:
+            return await self._help_specific(ctx, cmd_name)
+        await self._help_list(ctx)
+
+    async def _help_list(self, ctx: dis_snek.MessageContext) -> None:
+        cmds = await self._gather(ctx)
+
+        output = []
+        for cmd in cmds.values():
+            _temp = self._generate_command_string(cmd, ctx)
+            _temp += f"\n{cmd.brief}"
+
+            output.append(self._sanitise_mentions(_temp))
+        if len("\n".join(output)) > 500:
+            paginator = Paginator.create_from_list(self._client, output, page_size=500)
+            paginator.default_title = self.embed_title.format(username=self._client.user.username)
+            await paginator.send(ctx)
+        else:
+            embed = Embed(
+                title=self.embed_title.format(username=self._client.user.username),
+                description="\n".join(output),
+                color=dis_snek.BrandColors.BLURPLE,
+            )
+            await ctx.reply(embeds=embed)
+
+    async def _help_specific(self, ctx: dis_snek.MessageContext, cmd_name: str) -> None:
+        cmds = await self._gather(ctx)
+
+        if cmd := cmds.get(cmd_name.lower()):
```

Comment:
This is a naive approach and I *think* it won't work well with subcommands.

Reply:
I'll be honest, i completely forgot message subcommands were a thing

Reply:
![image](https://user-images.githubusercontent.com/22540825/161909249-ff870b72-a315-409e-bffb-20bc509b2084.png)
![image](https://user-images.githubusercontent.com/22540825/161909561-12a86920-1eda-4455-a87e-210b360ab5af.png)

seems to accidentally work

Reply:
Most likely works as `help._gather` generates a dict of `{cmd_resolved_name, cmd}` and the help_specific method is looking up by a resolved name (which the user would know commands by), so technically this should be the ideal route